enterprise/api/renderers.py

```diff
@@ -8,13 +8,18 @@
 from rest_framework.settings import api_settings
 
 
-class IgnoreClientContentNegotiation(DefaultContentNegotiation):
+class CustomContentNegotiation(DefaultContentNegotiation):
     """
-    Class for client content negotiation.
+    DRF was ignoring the order in render_classes, may be due to djangorestframework-xml library.
+    By Default, It was giving precedence to xml by default so we need to override the select_renderer method.1
     """
 
     def select_renderer(self, request, renderers, format_suffix):  # pylint: disable=signature-differs
-        # Allow URL style format override.  eg. "?format=json or http://example.com/v1/organizations.xml/
+        """
+        Allow URL style format override. Either you can specify the querystring parameter (/?format=json
+        or /?format=xml) or XML/JSON suffix at end of URL e.g. shttp://example.com/v1/organizations.xml/
+        """
+
```

Comment:
I still don't understand why this is needed. Why aren't we using [format_suffix_patterns](http://www.django-rest-framework.org/api-guide/format-suffixes/) to do this?

Reply:
Personally I dislike the format suffix, but that page does indicate support by default for [query params](http://www.django-rest-framework.org/api-guide/format-suffixes/#query-parameter-formats)

Based on the last paragraph on that page, it looks like the DRF have something against accept headers so I'm not surprised that isn't supported.

Reply:
Do we know that passing an appropriate accept header does not work? I would expect that it would.